### ISSUE TITLE
`eval (opam env)` in fish shell adds `.` to $HOME

### ISSUE THREAD

**Issue by goderich** [ocaml/opam#4418]
I've had this happen on two separate computers, and I believe I know the root cause of the issue. Running `opam env` in fish shell gives the following output:

```
(...other stuff...)
set -gx PATH ':/home/goderich/.opam/ocaml-system.4.11.1/bin' '/home/goderich/.local/bin' '/home/goderich/.emacs.d/bin' '/home/goderich/.bin' '/home/goderich/.config/bspwm/panel' '/home/goderich/.poetry/bin' '/usr/local/sbin' '/usr/local/bin' '/usr/bin/core_perl' '/usr/bin' '/usr/lib/jvm/default/bin' '/home/goderich/.local/bin';
```

It is that `:` at the beginning of the first argument that adds `.` to the environment. Running the command several times adds it as many times. Removing the colon gets rid of the problem (and removes already existing `.` from $PATH).

I think the colon is needed on POSIX shells (I'm not sure), while in fish it is not only unnecessary, but actively harmful. Having `.` in my $PATH led me on a merry bug chase for several days when trying to install Emacs from source.

**Comment by rjbou** [comment]
Is it the same problem than in #4403 ?
What is the version of opam you use ?

**Comment by goderich** [comment]
Oh my, I should have searched more thoroughly. That is indeed the same problem. I am on version `2.0.7` right now, I'll make sure to upgrade pronto. Thank you for the reply, I'm closing this (non-)issue.

**Comment by rjbou** [comment]
2.1.0 is not release (in beta versions), but i'll backport the fish fix in the next 2.0.8. Stay tuned for this one then ;)
